client/src/cli/apiToken.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  getAccessToken,
+  CantCreateRmmbrDir,
+  NoAccessToken,
+  NoHomeDirectory,
+} from "./configDir.ts";
+import { Result, ok, err, Unit } from "./deps.ts";
+
+class InvalidAccessToken extends Error {}
+
+export async function apiToken(): Promise<Result<string, Unit>> {
+  const accessToken = await getAccessToken();
+  if (accessToken.isErr) {
+    if (accessToken.error instanceof CantCreateRmmbrDir) {
+      console.error("Couldn't create the `.rmmbr` directory.");
+    } else if (accessToken.error instanceof NoHomeDirectory) {
+      console.error("Couldn't find your home directory.");
+    } else if (accessToken.error instanceof NoAccessToken) {
+      console.error("Not logged-in, run the `login` command first.");
+    }
+    return err();
+  }
+
+  return (await getOrCreateApiToken(accessToken.value)).match({
+    Ok: (apiToken) => {
+      console.log(apiToken);
+      return ok(apiToken);
+    },
+    Err: (e) => {
+      if (e instanceof InvalidAccessToken) {
+        console.log("Login expired, run the `login` command again.");
+      }
+      return err();
+    },
+  });
+}
+
+const apiTokenRequest = async (
+  accessToken: string,
+  method: "GET" | "POST" = "GET"
+) =>
+  await fetch("http://localhost:8000/api-token/", {
+    headers: { Authorization: `Bearer ${accessToken}` },
+    method,
+  });
+
+async function getOrCreateApiToken(
+  accessToken: string
+): Promise<Result<string, InvalidAccessToken>> {
+  const getResponse = await apiTokenRequest(accessToken);
+
+  if (getResponse.status == 200) {
+    return ok(await getResponse.text());
+  }
+
+  const createResponse = await apiTokenRequest(accessToken, "POST");
```

Comment:
this is a bit unclear to me, why would we try get and then post?

Reply:
First you see if there's an existing API token - if not, you create a new one.

Reply:
First you check if there's already an API token, and if not you create one